app/controllers/concerns/traceable.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'tracers/datadog_adapter'
+
+# Provides functionality to controllers for tagging them with specific services.
+# This tagging allows monitoring and tracing systems (like Datadog) to identify and group
+# the activities of specific controllers based on the service tags they are associated with.
+#
+# @example Setting a service tag for a controller
+#   class MyController < ApplicationController
+#     include Traceable
+#     service_tag :my_service
+#   end
+#
+# @note The actual interaction with the monitoring service is abstracted away
+#   and is handled by {Tracers::DatadogAdapter}. This module mainly focuses on
+#   providing the mechanism to set and validate the service tags for controllers.
+#
+# @see Tracers::DatadogAdapter for details on how tags are set for external services.
+module Traceable
+  extend ActiveSupport::Concern
+
+  # @!attribute [r] TRACE_ADAPTER
+  #   @return [Tracers::DatadogAdapter] Adapter for interacting with external tracing services.
+  TRACE_ADAPTER = Tracers::DatadogAdapter
```

Comment:
I wonder if this amount of indirection is unnecessary. My main goal as described in the issue was to avoid polluting _every single controller_ with `Datadog::Tracing` references, or any other vendor-specific reference.  Having the direct call to Datadog API in this single concern doesn't seem too bad.